### ISSUE TITLE
matlab crashes from running caffe.set_device(id) on windows

### ISSUE THREAD

**Issue by Kohit** [BVLC/caffe#5442]
Please use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) for usage, installation, or modeling questions, or other requests for help.
_Do not post such requests to Issues._ Doing so interferes with the development of Caffe.

Please read the [guidelines for contributing](https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md) before submitting this issue.

### Issue summary
When I try to use gpu on matcaffe, matlab crashes every time from running caffe.set_device(id), whether id is 0 or 1( I have one gpu: gtx 1050ti )
Where can I find the log file?( there is no matlab crash dump file in %temp% folder)

### Steps to reproduce

If you are having difficulty building Caffe or training a model, please ask the caffe-users mailing list. If you are reporting a build error that seems to be due to a bug in Caffe, please attach your build configuration (either Makefile.config or CMakeCache.txt) and the output of the make (or cmake) command.

### Your system configuration
Operating system:windows 10
Compiler:cl
CUDA version (if applicable):8.0
CUDNN version (if applicable):5.1
BLAS:
Python or MATLAB version (for pycaffe and matcaffe respectively):
matlab r2017a

**Comment by assafmus** [comment]
Matlab 2017 has issues with libraries provided with matlab.
This can cause conflicts with the ones actually used for caffe compilation.
Those libraries are placed in: <MATLAB>/bin/glnxa64 and <MATLAB>/sys/os/glnxa64
I've removed and replaced the ones provided by matlab. I have replaced libstdc++.so and removed all opencv libraries and libtiff.

Best way to verify that matlab has the correct libraries is to build caffe from within matlab. In matlab, go to the caffe folder and run "!make". Fix the errors until it successfully compiles.
